RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -815,6 +1073,22 @@ namespace lst {
     float y3 = mdsInGPU.anchorY[thirdMDIndex];
 
     circleRadius = computeRadiusFromThreeAnchorHits(acc, x1, y1, x2, y2, x3, y3, circleCenterX, circleCenterY);
+
+    if (not(passRZConstraint(acc,
+                             modulesInGPU,
+                             mdsInGPU,
+                             segmentsInGPU,
+                             innerInnerLowerModuleIndex,
+                             middleLowerModuleIndex,
+                             outerOuterLowerModuleIndex,
+                             firstMDIndex,
+                             secondMDIndex,
+                             thirdMDIndex,
+                             circleRadius,
+                             circleCenterX, 
+                             circleCenterY)))
```

Comment:
I moved it to the end because we need the circle fit to calculate the helix chi2. We can also moved both the circle fit and the r-z cuts to before the pointing cuts, but it wouldn't affect the physical performance because the cuts commute. I think it may affect the timing though. I'll do some tests to see which is the best order

Reply:
since the timing got worse; would it be practical to relax the original rz constraint (e.g. x1.5 or x2 looser cuts) keeping it linear (and fast) and then use the helix approximation later?

Reply:
Yes, we've decided to move the rz cuts before the pointing constraints, which improved the timing slightly